pkg/cmd/pulumi/crypto_http.go

```diff
@@ -42,6 +45,50 @@ func newServiceSecretsManager(s httpstate.Stack, stackName tokens.QName, configF
 	client := s.Backend().(httpstate.Backend).Client()
 	id := s.StackIdentifier()
 
+	// if there is no key OR the secrets provider is changing
+	// then we need to generate the new key based on the new secrets provider
+	if info.EncryptedKey == "" || info.SecretsProvider != secretsProvider {
+		dataKey, err := service.GenerateNewDataKey(id, client)
+		if err != nil {
+			return nil, err
+		}
+		info.EncryptedKey = base64.StdEncoding.EncodeToString(dataKey)
+
+		currentConfig := info.Config
+		if currentConfig.HasSecureValue() {
+			legacyCrypter, err := service.NewServiceSecretsManager(client, id, nil)
+			if err != nil {
+				return nil, err
+			}
+			legDecrypter, err := legacyCrypter.Decrypter()
+			if err != nil {
+				return nil, err
+			}
+			manager, err := service.NewServiceSecretsManager(client, id, dataKey)
+			if err != nil {
+				return nil, err
+			}
+			newEncrypter, err := manager.Encrypter()
+			if err != nil {
+				return nil, err
+			}
+			newProjectConfig, err := currentConfig.Copy(legDecrypter, newEncrypter)
+			if err != nil {
+				return nil, err
+			}
+			info.Config = newProjectConfig
+		}
```

Comment:
Needs a comment in the code but basically rewriting the config with the new encrypter